### PR TITLE
Introduce audit logs

### PR DESCRIPTION
**Pull request by prayerslayer**
Introducing a HTTP logger that sends audit events on `v1/put-tag` and `v2/put-manifest`, i.e. after uploading a tag. (Side note: Is v1 support still needed?)

The tests are passing, please review if everything makes sense.

### CODE REVIEW

File: src/org/zalando/stups/pierone/api_v1.clj [hunk 1]
```diff
@@ -130,6 +131,14 @@
       (try
         (sql/create-tag! params-with-user connection)
         (log/info "Stored new tag %s." params-with-user)
+        (let [tag-info {:team team
+                        :artifact artifact
+                        :tag name
+                        :repository (:repository api-config)}
+              scm-source (sql/get-scm-source
```
Comment (prayerslayer) [inline src/org/zalando/stups/pierone/api_v1.clj]: We should do `first` here.

**Comment by hjacobs** [comment]
@prayerslayer you could remove v1 code altogether, but this needs modifications for the tests.

I think we should remove v1 code now as it makes no sense to still have it and not include auditing there.

**Comment by prayerslayer** [comment]
There is also audit support in `v1`; we should probably remove the v1 code in a separate PR. (And maybe insert some logs to check if it is still called?)

**Comment by prayerslayer** [comment]
👍

**Comment by dryewo** [comment]
👍

**Pull request merged**